Blog: Cordova's Timeline for Node 6.x & 8.x Deprecation & Node 12.x Support (#965)
apache/cordova-docs: www/_posts/2019-04-11-nodejs-6.x-8.x-deprecation-timeline.md

Since October 30th, 2018, `node 10.x` has started its active Long Term Support ([LTS][1]) period. Thus, as `node 10.x` is the default download from [Node.js](https://nodejs.org/en/), we also recommend users to upgrade to the current LTS (10.x) before/by the time 8.x has reached EOL.

During these two distinct deprecation periods, we will prepare our tools, platforms, core libraries, and plugins to follow the Node.js Foundation reported EOL schedule.

We will also plan to add `node 12.x` support. Currently, `node 12.x` is pending for release on April 23rd, 2019.

JavaScript files in plugins themselves are unaffected since the JavaScript support for them is dependent on platform browser support. For the Electron platform, if using Node.js APIs there is a possibility of experience an effect.  

<!--more-->

## Cordova Timeline

### **March 2019**

* All AppVeyor/Travis CI builds will continue to test on Node.js 6.x, 8.x, and 10.x.

> ### **April 30th, 2019**

Author: ```suggestion
### **April 2019**
```

* All AppVeyor/Travis CI builds will test on Node.js 6.x, 8.x, 10.x, and 12.x.

> * Update requirement check and notify users that using Node.js 6.x, was deprecated by the Node team and is not recommended to use and should upgrade to the minimum LTS (10.x).

Author: ```suggestion
* Update requirement check and notify users: `Warning: Node.js 6.x has been deprecated. Please upgrade to the latest Node.js version available (LTS version recommended).`
```

> ### **September 2019**

Author: ```suggestion
### **Fall 2019**
```

> * Update requirement check and notify deprecation warning if the user is using Node.js 6.x and 8.x and recommend to upgrade to the minimum LTS (10.x).

Author: ```suggestion
* Update requirement check and notify users: `Warning: Node.js 8.x has been deprecated. Please upgrade to the latest Node.js version available (LTS version recommended).`
```

> ### **December 31st, 2019**

Author: ```suggestion
### **December 2019 / January 2020**
```

> * Deprecate use of Node.js 6.x

Reviewer: ```suggestion
* Deprecated use of Node.js 6.x
```

Reviewer 2: We should deprecate now, and remove support at the end of the year, or sooner for 6.

> * Deprecate use of Node.js 8.x

Reviewer: ```suggestion
* Deprecated use of Node.js 8.x
```

> * Officially Add support for Node.js 12.x

Author: ```suggestion
* Added Node.js 12.x  support.
```
